charts/agent/templates/_deployment-cluster-events-config.tpl

```diff
@@ -0,0 +1,271 @@
+  # 0.0.0.0 is hack for ipv6 on eks clusters
+  health_check:
+      endpoint: "0.0.0.0:13133"
+receivers:
+  # this is used to create a cluster resource by pulling namespaces and then dropping all but kube-system with filter processor
+  k8sobjects/cluster:
+    auth_type: serviceAccount
+    objects:
+      - name: namespaces
+        mode: pull
+        interval: 15m
+  # this pulls all objects listed below
+  k8sobjects/objects_pull:
+    auth_type: serviceAccount
+    objects:
+      ## P1
+      - {name: events, mode: pull, interval: 15m}
+      - {name: pods, mode: pull, interval: 15m}
+      - {name: namespaces, mode: pull, interval: 15m}
+      - {name: nodes, mode: pull, interval: 15m}
+      - {name: deployments, mode: pull, interval: 15m}
+      - {name: replicasets, mode: pull, interval: 15m}
+      - {name: configmaps, mode: pull, interval: 15m}
+      - {name: endpoints, mode: pull, interval: 15m}
+      ## P2
+      - {name: jobs, mode: pull, interval: 15m}
+      - {name: cronjobs, mode: pull, interval: 15m}
+      - {name: daemonsets, mode: pull, interval: 15m}
+      - {name: statefulsets, mode: pull, interval: 15m}
+      - {name: services, mode: pull, interval: 15m}
+      - {name: ingresses, mode: pull, interval: 15m}
+      - {name: secrets, mode: pull, interval: 15m}
+      - {name: persistentvolumeclaims, mode: pull, interval: 15m}
+      - {name: persistentvolumes, mode: pull, interval: 15m}
+      - {name: storageclasses, mode: pull, interval: 15m}
+      - {name: roles, mode: pull, interval: 15m}
+      - {name: rolebindings, mode: pull, interval: 15m}
+      - {name: clusterroles, mode: pull, interval: 15m}
+      - {name: clusterrolebindings, mode: pull, interval: 15m}
+      - {name: serviceaccounts, mode: pull, interval: 15m}
+  k8sobjects/objects_watch:
+    auth_type: serviceAccount
+    objects:
+      ## P1
+      - {name: events, mode: watch}
+      - {name: pods, mode: watch}
+      - {name: namespaces, mode: watch}
+      - {name: nodes, mode: watch}
+      - {name: deployments, mode: watch}
+      - {name: replicasets, mode: watch}
+      - {name: configmaps, mode: watch}
+      - {name: endpoints, mode: watch}
+      ## P2
+      - {name: jobs, mode: watch}
+      - {name: cronjobs, mode: watch}
+      - {name: daemonsets, mode: watch}
+      - {name: statefulsets, mode: watch}
+      - {name: services, mode: watch}
+      - {name: ingresses, mode: watch}
+      - {name: secrets, mode: watch}
+      - {name: persistentvolumeclaims, mode: watch}
+      - {name: persistentvolumes, mode: watch}
+      - {name: storageclasses, mode: watch}
+      - {name: roles, mode: watch}
+      - {name: rolebindings, mode: watch}
+      - {name: clusterroles, mode: watch}
+      - {name: clusterrolebindings, mode: watch}
+      - {name: serviceaccounts, mode: watch}
+processors:
+  # needs to come after transform/watch_objects in pipelines
+  observek8sattributes:
```

Comment:
I assume this is the new custom processor.
But yeah the current config.tpl format LGTM. Once we start iterating on it we can have more feedbacks